### PR TITLE
Update RFC 0096 (stacks removal) to match what landed in the spec

### PR DESCRIPTION
**Pull request by edmorley**
A few changes were made to what was in [RFC96](https://github.com/buildpacks/rfcs/blob/ce8991b3d739f763b8356ae8696ae582f18c4af6/text/0096-remove-stacks-mixins.md) when it came to update the spec:
https://github.com/buildpacks/spec/pull/365

This backports those changes to the RFC to try and prevent user confusion if they use the RFC as a means of learning about targets when implementing.

Specifically:
- The `io.buildpacks.distribution.*` Docker labels were renamed to `io.buildpacks.base.distro.*`.
- The `io.buildpacks.id` Docker label was renamed to `io.buildpacks.base.id`.
- The `buildpack.toml` table `[[targets.distributions]]` was renamed to `[[targets.distros]]` and the `versions` field within it renamed to `version` (along with its type changing from an array to a string).

See:
https://github.com/buildpacks/spec/blob/platform/0.13/platform.md#target-data

### CODE REVIEW

**Review by natalieparellano — COMMENTED** [review]
@edmorley thanks for this. Could you add the amended metadata at the bottom?

```
<!--
## Amended
### Meta
[meta-1]: #meta-1
- Name: (fill in the amendment name: Variable Rename)
- Start Date: (fill in today's date: YYYY-MM-DD)
- Author(s): (Github usernames)
- Amendment Pull Request: (leave blank)

### Summary

A brief description of the changes.

### Motivation

Why was this amendment necessary?
--->
```

**Comment by edmorley** [comment]
@natalieparellano Sure - added now :-)

**Comment by edmorley** [comment]
Oh I also found one more change that was made when the spec was updated, that I've backported now:
- The `versions` field within `[[targets.distros]]` was renamed to `version` (along with its type changing from an array to a string).

**Comment by edmorley** [comment]
> This backports those changes to the RFC to try and prevent user confusion if they use the RFC as a means of learning about targets when implementing.

To give some more context here - one of the reasons I look at the RFCs when implementing, is that they are quite prominently linked from the spec GitHub release entries:
https://github.com/buildpacks/spec/releases

File: text/0096-remove-stacks-mixins.md [hunk 1]
```diff
@@ -210,3 +210,23 @@ rename x86_64 -> amd64 in keeping with all other usages of arch. descriptors.
...
+- Name: Rename Docker labels and `buildpack.toml` table names
+- Start Date: 2024-04-08
+- Author(s): @edmorley
+- Amendment Pull Request: [rfcs#310](https://github.com/buildpacks/rfcs/pull/310)
+
+### Summary
+
+Changes were made to the Docker label and `buildpack.toml` table names between when this RFC was written and the changes were made to the spec in [spec#365](https://github.com/buildpacks/spec/pull/365), which have now been backported to the RFC:
+
+- The `io.buildpacks.distributions.*` Docker labels were renamed to `io.buildpacks.base.distros.*`.
```
Comment (edmorley) [inline text/0096-remove-stacks-mixins.md:226]:
```suggestion
- The `io.buildpacks.distributions.*` Docker labels were renamed to `io.buildpacks.base.distro.*`.
```

**Comment by edmorley** [comment]
(Me using the GitHub rebase button broke the GPG verification status of the commits, so I've squashed and manually pushed instead)

**Comment by edmorley** [comment]
Please can I have review/merge? :-)

**Review by natalieparellano — APPROVED** [review]

**Pull request merged**